### PR TITLE
10-CORE-GET-api-articles

### PR DESCRIPTION
**Pull request by miors**
endpoint /api/articles (topic query)
solution and test

### CODE REVIEW

File: models/articles.model.js [hunk 1]
```diff
@@ -12,15 +12,41 @@ exports.fetchArticleByID = (article_id) => {
...
+  }
+
+  const finalSqlArr = [];
+  let finalSqlStr = `
+  SELECT articles.author, title, articles.article_id, topic, articles.created_at, articles.votes, article_img_url, COUNT(comment_id)::int AS comment_count FROM articles LEFT JOIN comments ON articles.article_id  = comments.article_id
+  `;
+
+  if (topic) {
+    finalSqlStr += " WHERE topic=$1";
+    finalSqlArr.push(topic);
```
Comment (nc-hev-mag) [inline models/articles.model.js:42]: Great conditional building of SQL string

File: __tests__/integration.test.js [hunk 2]
```diff
@@ -114,6 +114,56 @@ describe("GET /api/articles", () => {
...
+          expect(typeof article.created_at).toBe("string");
+          expect(typeof article.votes).toBe("number");
+          expect(typeof article.article_img_url).toBe("string");
+          expect(typeof article.comment_count).toBe("number");
+        });
+      });
+  });
+
+  it("GET:200 should send empty array if topic query string has no match", () => {
+    return request(app)
```
Comment (nc-hev-mag) [inline __tests__/integration.test.js:142]: Great!

**Review by nc-hev-mag — COMMENTED** [review]
Brill Mior - just a small tweak to make in that first test!

File: __tests__/integration.test.js [hunk 3]
```diff
@@ -114,6 +114,56 @@ describe("GET /api/articles", () => {
...
+        expect(articles).toHaveLength(12);
+        expect(articles).toBeSortedBy("created_at", {
+          descending: true,
+        });
+        articles.forEach((article) => {
+          expect(Object.keys(article)).toHaveLength(8);
+          expect(typeof article.author).toBe("string");
+          expect(typeof article.title).toBe("string");
+          expect(typeof article.article_id).toBe("number");
+          expect(typeof article.topic).toBe("string");
```
[the remark this answers is not in the document]
Comment (miors) [reply __tests__/integration.test.js:132]: done

**Pull request merged**